Content.Server/Chat/Systems/SpeakOnTriggerSystem.cs

```diff
@@ -0,0 +1,44 @@
+using Content.Shared.Timing;
+using Robust.Shared.Prototypes;
+using Robust.Shared.Random;
+using Content.Server.Explosion.EntitySystems;
+
+namespace Content.Server.Chat.Systems;
+
+public sealed class SpeakOnTriggerSystem : EntitySystem
+{
+    [Dependency] private readonly UseDelaySystem _useDelay = default!;
+    [Dependency] private readonly IRobustRandom _random = default!;
+    [Dependency] private readonly IPrototypeManager _prototypeManager = default!;
+    [Dependency] private readonly ChatSystem _chat = default!;
+
+    public override void Initialize()
+    {
+        base.Initialize();
+        SubscribeLocalEvent<SpeakOnTriggerComponent, TriggerEvent>(OnTrigger);
+    }
+
+    private void OnTrigger(Entity<SpeakOnTriggerComponent> ent, ref TriggerEvent args)
+    {
+        if (args.Handled)
+            return;
+
+        TrySpeak(ent);
+    }
+
+    private void TrySpeak(Entity<SpeakOnTriggerComponent> ent)
+    {
```

Comment:
```suggestion
```
I had a look at the other `OnTrigger` functions and they don't seem to check for `args.Handled` which makes sense since sometimes you want multiple effects to happen. However do set `args.Handled = true` at the end because it is used to check if the trigger was successful.

Reply:
Makes sense!